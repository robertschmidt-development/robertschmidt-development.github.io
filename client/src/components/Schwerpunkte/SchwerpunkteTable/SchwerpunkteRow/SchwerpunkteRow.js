import React from 'react'

const SchwerpunkteRow = props => {

    const {row, onEdit, onDelete} = props

    return(
        <>
            <tr>
                <td>{row.todo}</td>
                <td>{row.baureihen}</td>
                <td>{row.baujahre}</td>
                <td>{row.bemerkung}</td>
                {/* <td>{row.bilder}</td> */}
                <td><button type="button" className="btn btn-outline-warning" onClick={() => onEdit(row)}>Edit</button></td>
                <td><button type="button" className="btn btn-outline-danger" onClick={() => onDelete(row)}>LÃ¶schen</button></td>
            </tr>
        </>
    )
}

export default SchwerpunkteRow