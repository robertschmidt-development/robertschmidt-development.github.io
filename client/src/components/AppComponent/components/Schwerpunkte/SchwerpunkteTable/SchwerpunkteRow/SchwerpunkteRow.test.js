import React from 'react'
import SchwerpunkteRow from './SchwerpunkteRow'
import {render, cleanup} from '@testing-library/react'

import renderer from 'react-test-renderer'

afterEach(cleanup)

// it('renders without crashing', () => {
//     const div = document.createElement('div')
//     const row = {todo: 'todo', baureihen: 'baureihen', baujahre: 'baujahre', bemerkung: 'bemerkung'}
//     const {getByTestId} = render(<SchwerpunkteRow row={row}/>, div)
//     expect(getByTestId('id')).toHaveContent('dxd')
// })

it('matches snapshot', () => {
    const row = {todo: 'todo', baureihen: 'baureihen', baujahre: 'baujahre', bemerkung: 'bemerkung'}
    const tree = renderer.create(<SchwerpunkteRow row={row}/>).toJSON()
    expect(tree).toMatchSnapshot()
})

it('matches snapshot 2', () => {
    const row = {todo: 'a', baureihen: 'b', baujahre: 'c', bemerkung: 'd'}
    const tree = renderer.create(<SchwerpunkteRow row={row}/>).toJSON()
    expect(tree).toMatchSnapshot()
})
