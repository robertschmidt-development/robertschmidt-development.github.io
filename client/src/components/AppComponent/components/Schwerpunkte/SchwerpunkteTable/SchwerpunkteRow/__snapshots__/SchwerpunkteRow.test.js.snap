// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<tr>
  <td>
    todo
  </td>
  <td>
    baureihen
  </td>
  <td>
    baujahre
  </td>
  <td>
    bemerkung
  </td>
  <td>
    <button
      className="btn btn-outline-warning"
      onClick={[Function]}
      type="button"
    >
      Edit
    </button>
  </td>
  <td>
    <button
      className="btn btn-outline-danger"
      onClick={[Function]}
      type="button"
    >
      Löschen
    </button>
  </td>
</tr>
`;

exports[`matches snapshot 2 1`] = `
<tr>
  <td>
    a
  </td>
  <td>
    b
  </td>
  <td>
    c
  </td>
  <td>
    d
  </td>
  <td>
    <button
      className="btn btn-outline-warning"
      onClick={[Function]}
      type="button"
    >
      Edit
    </button>
  </td>
  <td>
    <button
      className="btn btn-outline-danger"
      onClick={[Function]}
      type="button"
    >
      Löschen
    </button>
  </td>
</tr>
`;
